### PR TITLE
[Refactor][API2.0] API for TensorRT

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add new TensorRT API.

## Modification

requirement:

- [x] https://github.com/open-mmlab/mmdeploy/pull/419
- [x] https://github.com/open-mmlab/mmdeploy/pull/433

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tests/test_apis/test_torch2torchscript.py [hunk 1]
```diff
@@ -84,4 +84,5 @@ def test_torch2torchscript(input_name, output_name):
         model_cfg=get_model_cfg(),
         device='cpu')
 
+    print(ts_file)
```
Comment (AllentDan) [inline tests/test_apis/test_torch2torchscript.py:87]: is this for debugging?

File: mmdeploy/backend/tensorrt/utils.py [hunk 2]
```diff
@@ -10,38 +10,68 @@
...
+    """
+    load_tensorrt_plugin()
+    with trt.Logger() as logger, trt.Runtime(logger) as runtime:
+        with open(path, mode='rb') as f:
+            engine_bytes = f.read()
+        engine = runtime.deserialize_cuda_engine(engine_bytes)
+        return engine
+
+
+def from_onnx(onnx_model: Union[str, onnx.ModelProto],
```
Comment (AllentDan) [inline mmdeploy/backend/tensorrt/utils.py:41]: `from_onnx` sounds like an attribute function of some class. How about onnx2tensorrt_impl or keeping the original name?

Comment (grimoire) [reply mmdeploy/backend/tensorrt/utils.py:41]: `torch.from_numpy` is used to transform `ndarray` to `torch.Tensor`. This name follows the same rule.

File: tests/test_apis/test_torch2torchscript.py (hunk 1 above)
Comment (grimoire) [reply tests/test_apis/test_torch2torchscript.py]: has been fixed  in 419.

File: mmdeploy/apis/calibration.py [hunk 3]
```diff
@@ -1,107 +1,63 @@
...
-    """Create calibration table.
+@PIPELINE_MANAGER.register_pipeline()
+def create_calib_input_data(calib_file: str,
+                            deploy_cfg: Union[str, mmcv.Config],
+                            model_cfg: Union[str, mmcv.Config],
+                            model_checkpoint: Optional[str] = None,
+                            dataset_cfg: Optional[Union[str,
+                                                        mmcv.Config]] = None,
+                            dataset_type: str = 'val',
+                            device: str = 'cpu') -> None:
```
Comment (RunningLeon) [inline mmdeploy/apis/calibration.py:22]: may add docstring for this api function

File: mmdeploy/apis/utils/calibration.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                            inference_func: Optional[Callable] = None,
+                            model_partition: bool = False,
+                            context_info: Dict = dict(),
+                            device: str = 'cpu') -> None:
+    """Create calibration table.
+
+    Examples:
+        >>> from mmdeploy.apis import create_calib_input_data
+        >>> from mmdeploy.utils import get_calib_filename, load_config
+        >>> deploy_cfg = 'configs/mmdet/detection/' \
```
Comment (RunningLeon) [inline mmdeploy/apis/utils/calibration.py:28]: could align docstring style with https://github.com/open-mmlab/mmdeploy/pull/495

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**